packable/packable/src/packable/map.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types and implementations for packing and unpacking map values.
+
+extern crate alloc;
+
+#[cfg(feature = "usize")]
+use core::borrow::Borrow;
+use core::{convert::Infallible, fmt};
+#[cfg(feature = "std")]
+use std::{collections::HashMap, hash::Hash};
```

Comment:
Do we care about no_std for the types in iota-sdk? Because this might be problematic then
Would it make sense to use https://crates.io/crates/hashbrown instead/have an additional impl with it?

Reply:
Yeah you're right, I'll just use hashbrown

Reply:
Done